src/idlcxx/src/streamers.c

```diff
@@ -353,65 +395,51 @@ process_switch_type_spec(
   void *user_data)
 {
   static const char writefmt[] =
-    "  write(str, %1$s);\n"
-    "  switch(%1$s)\n{\n";
+    "  auto d = instance._d();\n"
+    "  write(str, d);\n";
   static const char readfmt[] =
-    "  %1$s %2$s;\n"
-    "  read(str, %2$s);\n"
-    "  switch(%2$s)\n{\n";
+    "  auto d = instance._d();\n"
+    "  read(str, d);\n";
   static const char movefmt[] =
-    "  move(str, %1$s);\n"
-    "  switch(%1$s)\n{\n";
+    "  auto d = instance._d();\n"
+    "  move(str, d);\n";
   static const char maxfmt[] =
-    "  max(str, %1$s);\n"
+    "  max(str, instance._d());\n"
     "  size_t union_max = str.position();\n";
```

Comment:
also create a holder for the "largest case" alignment
replace with "size_t union_max = str.position(), alignment_max = str.alignment();\n";